lib/handsoap/service.rb

```diff
@@ -265,7 +274,7 @@ def dispatch(doc, action)
         fire_on_log_header { "HandSoap Response [#{dispatch_id}]: length: [#{@http_client.body_str.length}], HTTP-Status: [#{@http_client.response_code}], Content-Type: [#{@http_client.content_type}]" }
         fire_on_log_body { Handsoap.pretty_format_envelope(@http_client.body_str) }
 
-        soap_response = Response.new(@http_client.body_str, envelope_namespace)
+        soap_response = Response.new(@http_client, envelope_namespace)
```

Comment:
So there seems to be some support for per-thread values (https://github.com/taf2/curb/blob/master/lib/curl.rb#L11), but not with the interface that we use (`Curl::Easy`)